### PR TITLE
fix autoresize of Entities.addEntity() and fix Quat.lookAtSimple()

### PR DESCRIPTION
**Pull request by AndrewMeadows**
* fix autoresize of Entities.addEntity() for a ModelEntityItem > 12m from camera
* fix Quat.lookAtSimple() to compare against absolute value of vector components for degenerate case

### CODE REVIEW

File: libraries/script-engine/src/Quat.cpp [hunk 1]
```diff
@@ -37,29 +37,30 @@ glm::quat Quat::lookAt(const glm::vec3& eye, const glm::vec3& center, const glm:
 glm::quat Quat::lookAtSimple(const glm::vec3& eye, const glm::vec3& center) {
     auto dir = glm::normalize(center - eye);
     // if the direction is nearly aligned with the Y axis, then use the X axis for 'up'
-    if (dir.x < 0.001f && dir.z < 0.001f) {
+    const float MAX_ABS_Y_COMPONENT = 0.9999991f;
```
Comment (ZappoMan) [inline libraries/script-engine/src/Quat.cpp]: you should make a different PR for this... this is unrelated - right?

Comment (AndrewMeadows) [inline libraries/script-engine/src/Quat.cpp]: Yes, it is unrelated but I could not justify making a distinct PR for it.

**Comment by hifi-gustavo** [comment]
Build console output for commit: b7c692254515cf0a82e5efc7a5ece646f0c6f18e available at: https://gist.github.com/eb515777251e5b36f4fa6fef09859892<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8992-b7c6922.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8992-b7c6922.dmg

File: libraries/script-engine/src/Quat.cpp (hunk 1 above)
Comment (ZappoMan) [inline libraries/script-engine/src/Quat.cpp]: ok - updated the title of the PR

Comment (ZappoMan) [inline libraries/script-engine/src/Quat.cpp]: make sure your test plan includes both.

**Comment by AndrewMeadows** [comment]
This PR fixes two issues:
https://github.com/highfidelity/hifi/issues/7848
https://github.com/highfidelity/hifi/issues/8964

To test:

(1) For issue 7848 run this script:
```
// Demonstrate model entities not auto-resizing if rezzed 13 metres away from the camera position.
var distance = 13;  
var modelURL = "https://s3.amazonaws.com/hifi-public/marketplace/hificontent/ozan/PalmTree.fbx";
var position = Vec3.sum(Camera.position, Vec3.multiply(distance, Vec3.multiplyQbyV(Camera.orientation, Vec3.UNIT_NEG_Z)));
var properties = {
    type: "Model",
    modelURL: modelURL,
    position: position,
    name: "PalmTree"
};
Entities.addEntity(properties);

```
In bugged Interface the PalmTree will not show up immediately, but it will if you move the camera forward a little bit.
In this PR the PalmTree will show up immediately.

(2) For issue 8964 run this script:
TODO: finish this test...

**Comment by OmegaHeron** [comment]
Testing Windows 10

QA Pass

Compared to current RC (25), RC misbehaves while this PR gives advertised results.
